app/lib/nhs/immunisations_api.rb

```diff
@@ -1,7 +1,14 @@
 # frozen_string_literal: true
 
 module NHS::ImmunisationsAPI
-  PROGRAMME_TYPES = %w[flu hpv].freeze
+  WRITE_PROGRAMME_TYPES = %w[flu hpv].freeze
+  READ_PROGRAMME_TYPES = %w[flu].freeze
```

Comment:
I wonder if these might be nicer as methods on the `Programme` model: `programme.can_read_from_nhs_immunisations_api?` for example.

It would mean further down you could replace:

```rb
!programmes.all? do |programme|
  programme.type.in?(READ_PROGRAMME_TYPES)
end
```

with:

```rb
!programmes.all?(&:can_read_from_nhs_immunisations_api?)
```

which I think is clearer.